src/Sylius/Bundle/CoreBundle/Resources/config/validation/PromotionRule.xml

```diff
@@ -107,26 +107,27 @@
             <constraint name="Sylius\Bundle\CoreBundle\Validator\Constraints\ChannelCodeCollection">
                 <option name="allowExtraFields">true</option>
                 <option name="groups">sylius_promotion_rule_total_of_items_from_taxon</option>
+                <option name="validateAgainstAllChannels">true</option>
                 <option name="constraints">
                     <constraint name="Collection">
                         <option name="allowExtraFields">true</option>
+                        <option name="groups">sylius_promotion_rule_total_of_items_from_taxon</option>
                         <option name="fields">
                             <value key="amount">
-                                <constraint name="Required">
-                                    <constraint name="NotBlank">
-                                        <option name="groups">sylius_promotion_rule_total_of_items_from_taxon</option>
-                                    </constraint>
-                                    <constraint name="Type">
-                                        <option name="type">numeric</option>
-                                        <option name="groups">sylius_promotion_rule_total_of_items_from_taxon</option>
-                                    </constraint>
+                                <constraint name="NotBlank">
+                                    <option name="groups">sylius_promotion_rule_total_of_items_from_taxon</option>
+                                </constraint>
+                                <constraint name="Type">
+                                    <option name="type">numeric</option>
+                                    <option name="groups">sylius_promotion_rule_total_of_items_from_taxon</option>
```

Comment:
Is it possible to extract changes to cart promotions from this PR (by possible, I mean quickly and easily)? 🤔

Reply:
I extracted most of them to separate [PR](https://github.com/Sylius/Sylius/pull/15756/files)